app/src/main/java/ardjomand/leonardo/arcmovies/moviedetails/MovieDetailsPresenter.java

```diff
@@ -0,0 +1,49 @@
+package ardjomand.leonardo.arcmovies.moviedetails;
+
+import ardjomand.leonardo.arcmovies.Utils;
+import ardjomand.leonardo.arcmovies.data.MoviesRepository;
+import ardjomand.leonardo.arcmovies.model.MovieDetails;
+import ardjomand.leonardo.arcmovies.model.UpcomingMovie;
+
+/**
+ * Created by unity on 25/02/18.
+ */
+
+class MovieDetailsPresenter implements MovieDetailsContract.Presenter {
+
+    private final MovieDetailsContract.View mView;
+    private final MoviesRepository mRepository;
+
+    MovieDetailsPresenter(MovieDetailsContract.View view, MoviesRepository moviesRepository) {
+        mView = view;
+        mView.setPresenter(this);
+        mRepository = moviesRepository;
+    }
+
+    @Override
+    public void getMovieDetails(int movieId) {
+        mView.setLoading(true);
+        mRepository.loadMovieDetails(new MoviesRepository.LoadMovieDetailsCallback() {
+            @Override
+            public void onSuccess(MovieDetails movieDetails) {
+                mView.setLoading(false);
+                mView.showMovieDetails(movieDetails.getTitle(), movieDetails.getBackdropPath(),
+                        movieDetails.getOverview(), Utils.getGenreNames(movieDetails.getGenres()),
+                        movieDetails.getReleaseDate());
+            }
+
+            @Override
+            public void onFailure(String message) {
+                mView.setLoading(false);
+                mView.showErrorMessage(message);
+            }
+        }, movieId);
+    }
+
+    @Override
+    public void getMovieDetails(UpcomingMovie upcomingMovie) {
+        mView.showMovieDetails(upcomingMovie.getTitle(), upcomingMovie.getBackdropPath(),
+                upcomingMovie.getOverview(), upcomingMovie.getGenreNames(),
+                upcomingMovie.getReleaseDate());
```

Comment:
The release date that comes from the API is not really fit for presenting to the user, is it?

Reply:
I will display the data using the device locale as shown on this post:
https://stackoverflow.com/questions/11093182/date-formatting-based-on-user-locale-on-android